src/libostree/ostree-repo-pull.c

```diff
@@ -1827,6 +1938,86 @@ scan_commit_object (OtPullData                 *pull_data,
     }
 #endif /* OSTREE_DISABLE_GPGME */
 
+  if (pull_data->sign_verify &&
+      !g_hash_table_contains (pull_data->verified_commits, checksum))
+    {
+      gboolean ret = FALSE;
+      /* list all signature types in detached metadata and check if signed by any? */
+      g_auto (GStrv) names = ostree_sign_list_names();
+      for (guint i=0; i < g_strv_length (names); i++)
```

Comment:
The strv ends with a NULL, so you can do : 
`for (guint i=0; names[i] != NULL; i++)`
That save some calls to g_strv_length()
There are other places to do that too.

Reply:
prefer the current way -- looks a bit clearer for me with minimal overhead for.
And `g_strv_length()` is used widely in other places, so do not see why need to use other approach.

Reply:
The compiler can't know that it will be constant during the loop, so we'll evaluate it each time.  This doesn't matter too much *here* but having apparently `O(N)` code actually be `O(N²)` can be quite surprising if code is moved into an even mildly performance sensitive context.

Best to use an iterator, which we do elsewhere:

```
for (char ** iter = names; iter && *iter; iter++)
{
  const char *name = *iter;
  ...
}
```

Reply:
Just because the amount of elements _here_ is not expected to be big for years, I thought to leave it as is.
But now carefully re-read that code -- agree, a bit stupid thing I wrote here. Fixed.